### PR TITLE
updating Bio.File for python3

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request updates Bio.File for python3.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Looks like nothing clashes with #2523 which is good (could you take a look at that?).

You are doing quite a lot in this one pull request - minor cleanup, dictionary inheritance changes (which makes sense to me), and SQLite changes (why use ``sqlite3.dbapi2``?).

**Comment by mdehoon** [comment]
> Looks like nothing clashes with #2523 which is good (could you take a look at that?).
> 
> You are doing quite a lot in this one pull request - minor cleanup, dictionary inheritance changes (which makes sense to me), and SQLite changes (why use `sqlite3.dbapi2`?).

The SQLite code didn't actually change, I just removed the `_sqlite` obfuscation.

File: Bio/File.py [hunk 1]
```diff
@@ -472,10 +332,10 @@ def _load_index(self):
         index_filename = self._index_filename
         relative_path = self._relative_path
         filenames = self._filenames
-        format = self._format
+        fmt = self._format
         proxy_factory = self._proxy_factory
 
-        con = _sqlite.connect(index_filename, check_same_thread=False)
+        con = sqlite3.dbapi2.connect(index_filename, check_same_thread=False)
```
Comment (peterjc) [inline Bio/File.py]: Why ``sqlite3.dbapi2.connect`` versus ``sqlite3.connect``?

Comment (mdehoon) [reply Bio/File.py]: original code has
```
     from sqlite3 import dbapi2 as _sqlite
```

Comment (peterjc) [reply Bio/File.py]: Hah. As you said, obfuscated.

**Comment by mdehoon** [comment]
thanks

**Pull request merged**